packages/ra-ui-materialui/src/list/datagrid/DatagridBody.tsx

```diff
@@ -57,6 +58,8 @@ const DatagridBody: FC<DatagridBodyProps> = React.forwardRef(
                         record: data[id],
                         resource,
                         rowClick,
+                        expandable:
```

Comment:
With this implementation expand prop gets passed down to every DatagridRow regardless of the value of isRowExpandable.
If this is not ok, I could change it.
Now that I think about it is much more clean with a code like:

expand: (!isRowExpandable || isRowExpandable(data[id])) && expand, in the DatagridBody, thus leaving untouched DatagridRow component and avoiding adding the expandable prop.